src/responder/ssh/ssh_cmd.c

```diff
@@ -123,11 +124,120 @@ static errno_t ssh_cmd_get_user_pubkeys(struct cli_ctx *cli_ctx)
     return ret;
 }
 
+struct priv_sss_debug {
+    int level;
+};
+
+static void ssh_ext_debug(void *private, const char *file, long line,
+                      const char *function, const char *format, ...)
+{
+    va_list ap;
+    struct priv_sss_debug *data = private;
+    int level = SSSDBG_OP_FAILURE;
+
+    if (data != NULL) {
+        level = data->level;
+    }
+
+    if (DEBUG_IS_SET(level)) {
+        va_start(ap, format);
+        sss_vdebug_fn(file, line, function, level, APPEND_LINE_FEED,
+                      format, ap);
+        va_end(ap);
+    }
+}
+
+static errno_t ssh_cmd_refresh_certmap_ctx(struct ssh_ctx *ssh_ctx,
+                                           struct sss_domain_info *domains)
+{
+
+    struct sss_certmap_ctx *sss_certmap_ctx = NULL;
+    struct sss_domain_info *dom;
+    struct certmap_info **certmap_list;
+    size_t c;
+    int ret;
+    bool rule_added;
+
+    if (!ssh_ctx->use_cert_keys
+            || ssh_ctx->certmap_last_read
+                    >= ssh_ctx->rctx->get_domains_last_call.tv_sec) {
+        DEBUG(SSSDBG_TRACE_ALL, "No certmap update needed.\n");
+        return EOK;
+    }
+
+    ret = sss_certmap_init(ssh_ctx, ssh_ext_debug, NULL, &sss_certmap_ctx);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "sss_certmap_init failed.\n");
+        goto done;
+    }
+
+    rule_added = false;
+    DLIST_FOR_EACH(dom, domains) {
+        certmap_list = dom->certmaps;
+        if (certmap_list == NULL || *certmap_list == NULL) {
+            continue;
+        }
+
+        for (c = 0; certmap_list[c] != NULL; c++) {
+
```

Comment:
could you remove this empty line please

Reply:
I would prefer to keep it, because I think it helps to read the code especially with the additional empty line you suggested below.